File: backend/metier_generator/templates/threed/api/pagination/pagination.js
// Middleware de pagination API threed (Node.js) - Ultra avancé
// Gère la validation, les bornes, la doc, l'intégration métier
// Utilisation : app.use(require('./pagination'));

function paginationMiddleware(req, res, next) {
  let page = parseInt(req.query.page || '1', 10);
  let size = parseInt(req.query.size || '10', 10);
  if (isNaN(page) || page < 1) page = 1;
  if (isNaN(size) || size < 1 || size > 100) size = 10;
  req.pagination = { page, size };
  // Expose pour la logique métier
  req.businessContext = req.businessContext || {};
  req.businessContext.pagination = req.pagination;
  next();
}

module.exports = paginationMiddleware;

// Exemple d'intégration :
// app.use(require('./pagination'));
// Dans un contrôleur : req.pagination.page, req.pagination.size

File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/api/pagination/pagination.py
"""
Middleware de pagination API threed (Python) - Ultra avancé
Gère la validation, les bornes, la doc, l'intégration métier
Utilisation : app.before_request(pagination_middleware)
"""
from flask import g, request


def pagination_middleware():
    try:
        page = int(request.args.get('page', 1))
        size = int(request.args.get('size', 10))
    except Exception:
        page, size = 1, 10
    if page < 1:
        page = 1
    if size < 1 or size > 100:
        size = 10
    g.pagination = {'page': page, 'size': size}
    if not hasattr(g, 'business_context'):
        g.business_context = {}
    g.business_context['pagination'] = g.pagination
